p2p/node/peerManager/peerManager.go

```diff
@@ -403,6 +397,21 @@ func (pm *BasicPeerManager) removePeerFromTopic(peerID p2p.PeerID, topicStr stri
 	return nil
 }
 
+// Looks up which topics a peer participates in
+func (pm *BasicPeerManager) getPeerTopics(peerID p2p.PeerID) []string {
+	key := datastore.NewKey(peerID.String())
+	qualities := []PeerQuality{Best, Responsive, LastResort}
+	topics := []string{}
+	for topic, dbs := range pm.peerDBs {
+		for quality := range qualities {
+			if exists, _ := dbs[quality].Has(pm.ctx, key); exists {
```

Comment:
what if the db doesnt have an entry for this quality? wouldnt `dbs[quality]` crash?

Reply:
we will always have a DB for each quality level. If we failed to setup our DBs correctly, then a crash is appropriate.